Doc/reference/expressions.rst

```diff
@@ -406,7 +406,8 @@ brackets or curly braces.
 Variables used in the generator expression are evaluated lazily when the
 :meth:`~generator.__next__` method is called for the generator object (in the same
 fashion as normal generators).  However, the iterable expression in the
-leftmost :keyword:`!for` clause is immediately evaluated, so that an error
+leftmost :keyword:`!for` clause is immediately evaluated, and the
```

Comment:
What about async for?

Reply:
It is the same. `async for` is mentioned at the end of this section. What is written here applies to synchronous and asynchronous generator expressions (which are created not only by `async for`).